plugins/kinetic-dp/fu-kinetic-dp-aux-isp.c

```diff
@@ -0,0 +1,288 @@
+    {FU_KINETIC_DP_CHIP_PUMA_2900, FU_KINETIC_DP_FW_STATE_APP, {'M', 'C', '2', '9', '0'}, 5},
+    {FU_KINETIC_DP_CHIP_PUMA_2900, FU_KINETIC_DP_FW_STATE_APP, {'M', 'C', '2', '9', '1', '0'}, 6}};
+
+const gchar *
+fu_kinetic_dp_aux_isp_get_chip_id_str(FuKineticDpChip chip_id)
+{
+	if (chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000)
+		return "KTM50X0";
+	if (chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200)
+		return "KTM52X0";
+	if (chip_id == FU_KINETIC_DP_CHIP_PUMA_2900)
+		return "MC2900";
+	return "";
+}
+
+static gboolean
+fu_kinetic_dp_aux_isp_get_basic_dev_info_from_branch_id(const guint8 *br_id_str_buf,
+							const guint8 br_id_str_buf_size,
+							FuKineticDpDevInfo *dev_info,
+							GError **error)
+{
+	guint32 num = G_N_ELEMENTS(kt_dp_branch_dev_info_table);
+	g_autofree gchar *str = NULL;
+
+	g_return_val_if_fail(br_id_str_buf != NULL, FALSE);
+	g_return_val_if_fail(br_id_str_buf_size >= DPCD_SIZE_BRANCH_DEV_ID_STR, FALSE);
+	g_return_val_if_fail(dev_info != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	dev_info->chip_id = FU_KINETIC_DP_CHIP_NONE;
+	dev_info->fw_run_state = FU_KINETIC_DP_FW_STATE_NONE;
+
+	/* find the device info by branch ID string */
+	for (guint32 i = 0; i < num; i++) {
+		if (memcmp(br_id_str_buf,
+			   kt_dp_branch_dev_info_table[i].id_str,
+			   kt_dp_branch_dev_info_table[i].str_len) == 0) {
+			/* found the chip in the table */
+			dev_info->chip_id = kt_dp_branch_dev_info_table[i].chip_id;
+			dev_info->fw_run_state = kt_dp_branch_dev_info_table[i].fw_run_state;
+			return TRUE;
+		}
+	}
+
+	/* there might not always be nul character in the DPCD branch ID string */
+	str = g_strndup((const gchar *)br_id_str_buf, DPCD_SIZE_BRANCH_DEV_ID_STR);
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_INTERNAL,
+		    "%s is not a supported Kinetic device",
+		    str);
+	return FALSE;
+}
+
+gboolean
+fu_kinetic_dp_aux_isp_enable_aux_forward(FuKineticDpConnection *connection,
+					 FuKineticDpChip root_dev_chip_id,
+					 FuKineticDpFwState root_dev_state,
+					 FuKineticDpDev target_port,
+					 GError **error)
+{
+	if (root_dev_state != FU_KINETIC_DP_FW_STATE_APP) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "host device [%s %s] doesn't support to enable AUX forwarding",
+			    fu_kinetic_dp_aux_isp_get_chip_id_str(root_dev_chip_id),
+			    fu_kinetic_dp_fw_state_to_string(root_dev_state));
+		return FALSE;
+	}
+
+	if (root_dev_chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000 ||
+	    root_dev_chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200) {
+		if (!fu_kinetic_dp_secure_aux_isp_enable_aux_forward(connection,
+								     target_port,
+								     error)) {
+			g_prefix_error(error, "failed to enable AUX forwarding: ");
+			return FALSE;
+		}
+		/* Wait 10ms for host to process AUX forwarding command */
+		g_usleep(10 * 1000);
+		return TRUE;
+	}
+
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_INTERNAL,
```

Comment:
isn't this a `G_IO_ERROR`?  Doesn't seem like it's fwupd domain is appropriate

Reply:
done, thanks.